resources/assets/js/helpers/router.ts

```diff
@@ -37,17 +37,21 @@ export const useUrlHash = (): void => {
 };
 
 export const useRouter = (
-  routes: Routes<any, React.ReactElement>,
+  routes: Routes<any, any>,
 ): React.ReactElement | null => {
   const location = useLocation();
   const router = useMemo(() => new UniversalRouter(routes), [routes]);
   const [component, setComponent] = useState<React.ReactElement | null>(null);
 
   // Render the result of routing
   useEffect((): void => {
-    router
-      .resolve(location.pathname)
-      .then((result): void => setComponent(result));
+    router.resolve(location.pathname).then((result): void => {
+      const { defaultMessage } = result.title.props;
```

Comment:
You're passing in an entire `FormattedMessage` React element in as `result.title`, but you're only using the props of that react element. You might as well just pass the props directly.

Actually, You're only grabbing defaultMessage from those props, which is always the English string (it doesn't get translated). I think you should try to translate the message before it gets here, so that title can be a simple string.